File: src/messages/message.ts
/**
 * 会话消息模型定义。
 *
 * @author 真心
 */

/**
 * 模型触发的工具调用。
 */
export interface ToolCall {
  readonly id: string;
  readonly name: string;
  readonly argumentsJson: string;
}

/**
 * 用户消息。
 */
export interface UserMessage {
  readonly role: "user";
  readonly content: string;
}

/**
 * 助手消息。
 */
export interface AssistantMessage {
  readonly role: "assistant";
  readonly content: string;
  readonly toolCalls: readonly ToolCall[];
}

/**
 * 工具结果消息。
 */
export interface ToolResultMessage {
  readonly role: "tool";
  readonly toolCallId: string;
  readonly toolName: string;
  readonly content: string;
  readonly isError: boolean;
}

/**
 * banka 内部会话消息联合类型。
 */
export type ConversationMessage = UserMessage | AssistantMessage | ToolResultMessage;

File: internal/messages/message.go
// Package messages defines the internal conversation model.
package messages

// ToolCall is a model requested tool invocation.
type ToolCall struct {
	ID            string
	Name          string
	ArgumentsJSON string
}

// Message is one conversation item.
type Message struct {
	Role       string
	Content    string
	ToolCalls  []ToolCall
	ToolCallID string
	ToolName   string
	IsError    bool
}

// NewUserMessage creates a user message.
func NewUserMessage(content string) Message {
	return Message{Role: "user", Content: content}
}

// NewAssistantMessage creates an assistant message.
func NewAssistantMessage(content string, toolCalls []ToolCall) Message {
	return Message{Role: "assistant", Content: content, ToolCalls: toolCalls}
}

// NewToolMessage creates a tool result message.
func NewToolMessage(toolCallID string, toolName string, content string, isError bool) Message {
	return Message{
		Role:       "tool",
		ToolCallID: toolCallID,
		ToolName:   toolName,
		Content:    content,
		IsError:    isError,
	}
}
